DOCUMENTATION.md

```diff
@@ -0,0 +1,115 @@
+# Provider Documentation
+
+## Building and Reviewing
+
+The documentation is built from templates stored in the `templates`
+folder, along with examples stored in the `examples` folder. All of
+this content is copied to the `docs` folder, with the templates fully
+rendered, when the documentation is built.
+
+To validate the `templates` directory structure and its references to
+schemas in the provider and to examples, run:
+
+```shell
+make validate-docs
+```
+
+To build the documentation, replacing all of the contents of the
+`docs` folder:
+
+```shell
+make generate-docs
+```
+
+To preview a single documentation file, copy the content of the
+generated Markdown file into the [preview
+tool](https://registry.terraform.io/tools/doc-preview) on the
+HashiCorp website.
+
+## Templates
+
+Every resource and data source in the provider has its own
+documentation, which includes details of making use of that
+resource/data source, along with the details of its schema.
+
+If you do not write a template for a resource or data source that you
+create, `make generate-docs` will use a basic template when it
+generates documentation for the provider. This template will include
+only the name of the resource/data source and the details of its
+schema. For resources in particular, this is insufficient to guide
+users as it does not include examples of HCL which make use of the
+resource, or instructions for how to import an existing resource into
+the Terraform state.
+
+Note: The documentation generation tool gathers the list of known data
+sources and resources from the provider itself, not by scanning the
+source directory. As a result, if you have created a new data source
+or resource but you have not added into the top-level schema in
+[provider.go](fastly/provider.go) it will not be included in the
+generated documentation. If you create a template for the new data
```

Comment:
"If you create a template for the new data source or resource..."
(Fixes subject-verb agreement.)